OpenRA.Mods.Common/Widgets/Logic/Ingame/AddFactionSuffixLogic.cs

```diff
@@ -24,17 +24,29 @@ public AddFactionSuffixLogic(Widget widget, World world)
 				faction = world.LocalPlayer.Faction.InternalName;
 			var suffix = "-" + faction;
 
-			var buttonWidget = widget as ButtonWidget;
-			if (buttonWidget != null)
+			if (widget as ButtonWidget != null)
+			{
+				var buttonWidget = widget as ButtonWidget;
```

Comment:
These lines can then simplify to a direct cast and set: `((ButtonWidget)widget).Background += suffix;`

Reply:
Fixed.